### PR TITLE
Implement LearningObjectGateway for FileMetadata module

### PR DESCRIPTION
**Pull request by gshaw1997**
This PR implements a `ModuleLearningObjectGateway` which proxies request through the `LearningObjectAdapter`. This Gateway allows the `FileMetadata` module to fetch data about Learning Objects.

### CODE REVIEW

File: src/FileMetadata/drivers/LearningObjectGateway/ModuleLearningObjectGateway.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { LearningObjectGateway } from '../../interfaces';
+import { Requester, LearningObjectSummary } from '../../typings';
+import { LearningObjectAdapter } from '../../../LearningObjects/LearningObjectAdapter';
+
+export class ModuleLearningObjectGateway extends LearningObjectGateway {
```
Comment (DonSeannelly) [inline src/FileMetadata/drivers/LearningObjectGateway/ModuleLearningObjectGateway.ts]: Can you elaborate on why you made the design decision to create this proxy?

File: src/FileMetadata/interfaces/LearningObjectGateway.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { LearningObjectSummary, Requester } from '../typings';
+
+export abstract class LearningObjectGateway {
```
Comment (DonSeannelly) [inline src/FileMetadata/interfaces/LearningObjectGateway.ts]: Why did you go with an abstract class here?

Comment (gshaw1997) [reply src/FileMetadata/interfaces/LearningObjectGateway.ts]: The decision to go with an abstract class is discussed here.
https://github.com/Cyber4All/learning-object-service/pull/337#discussion_r290781423

**TLDR;** Pure abstract classes as interfaces are useful because their names can be used as DI tokens.

File: src/FileMetadata/drivers/LearningObjectGateway/ModuleLearningObjectGateway.ts (hunk 1 above)
Comment (gshaw1997) [reply src/FileMetadata/drivers/LearningObjectGateway/ModuleLearningObjectGateway.ts]: The business logic of this module only needs a small subset of the features provided by the `LearningObjectAdapter` and has its needs defined by the `LearningObjectGateway` interface. This abstraction allows us to swap out `LearningObjectGateway` implementations without having to modify calls from our Interactor.

Comment (DonSeannelly) [reply src/FileMetadata/drivers/LearningObjectGateway/ModuleLearningObjectGateway.ts]: I definitely like the pattern of keeping interfaces small and scoped to their use. I can't say I'm satisfied with the names, but also can't put my finger on why. I'll have to think it over some more - for now, I'm happy with this.
